definitions/scenarios/upgrade_to_foreman_3_3.rb

```diff
@@ -0,0 +1,91 @@
+module Scenarios::Foreman_3_3
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          !feature(:satellite) && feature(:foreman)
+          (feature(:foreman_server).current_minor_version == '3.2' || \
+          ForemanMaintain.upgrade_in_progress == '3.3')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '3.3'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Foreman 3.3'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+    end
+  end
+
+  class PreMigrations < Abstract
```

Comment:
Apologies if this is nomenclature that is already used throughout, and consider this more like educating me. There is 'Upgrade' lingo used in these procedures, e.g. `PreUpgradeCheck` and then there is Migration lingo used such as this step. 

What in this context does upgrade vs migration mean?

Reply:
I agree, the different steps should be indeed like,
~~~
PreUpgradeChecks
PreUpgradeProcedures
Upgrade
PostUpgradeProcedures
PostUpgradeChecks
~~~
As per these the descriptions should be updated as well. These were introduced with upgrades feature in early stage and being used as is, maybe no one questioned about the confusions.